org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -275,7 +278,7 @@ static ICompilationUnit getFakeCompilationUnit(URI uri, IProgressMonitor monitor
 		IProject project = JavaLanguageServerPlugin.getProjectsManager().getDefaultProject();
 		if (project == null || !project.isAccessible()) {
 			String fileName = path.getFileName().toString();
-			if (fileName.endsWith(".java") || fileName.endsWith(".class")) {
+			if (JDTUtils.isJavaFile(fileName) || fileName.endsWith(".class")) {

```

Comment:
I don't think you need to qualify with `JDTUtils` here.

Reply:
Fixed.